Lib/test/test__xxsubinterpreters.py

```diff
@@ -393,7 +393,19 @@ def test_bytes(self):
                             for i in range(-1, 258))
 
     def test_int(self):
-        self._assert_values(range(-1, 258))
+        self._assert_values(itertools.chain(range(-1, 258),
+                                            [sys.maxsize, -sys.maxsize - 1]))
+
+    def test_non_shareable_int(self):
```

Comment:
What is the result of this test on 32-bit platforms without the `pystate.c` changes?

Reply:
```
======================================================================
FAIL: test_non_shareable_int (test.test__xxsubinterpreters.ShareableTypeTests) [2147483648]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch2/izbyshev/cpython/Lib/test/test__xxsubinterpreters.py", line 408, in test_non_shareable_int
    interpreters.channel_send(self.cid, i)
AssertionError: OverflowError not raised

======================================================================
FAIL: test_non_shareable_int (test.test__xxsubinterpreters.ShareableTypeTests) [-2147483649]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch2/izbyshev/cpython/Lib/test/test__xxsubinterpreters.py", line 408, in test_non_shareable_int
    interpreters.channel_send(self.cid, i)
AssertionError: OverflowError not raised

----------------------------------------------------------------------
```

And if I change `test_int` to use `sys.maxsize + 1` and `-sys.maxsize - 2`:
```
======================================================================
FAIL: test_int (test.test__xxsubinterpreters.ShareableTypeTests) [2147483648]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch2/izbyshev/cpython/Lib/test/test__xxsubinterpreters.py", line 370, in _assert_values
    self.assertEqual(got, obj)
AssertionError: -2147483648 != 2147483648

======================================================================
FAIL: test_int (test.test__xxsubinterpreters.ShareableTypeTests) [-2147483649]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/scratch2/izbyshev/cpython/Lib/test/test__xxsubinterpreters.py", line 370, in _assert_values
    self.assertEqual(got, obj)
AssertionError: 2147483647 != -2147483649
```

Reply:
Thanks for checking that.